### DISCUSSION TITLE
How to write a build script to change framework from "Do not Embed" to "Embed and sign"

### DISCUSSION THREAD

**Discussion by majeed-platogo** [apache/cordova#477] [Q&A] [answered]
Hi all,

I'm trying to use a custom plugin in the plugins `plugin.xml`, I have the following line
```xml
<source-file src="src/ios/bass.xcframework" target-dir="lib" framework="true" />
```
when the build is completed the app is stuck on the splashscreen.

But, if i navigate `projectName -> General -> Frameworks, Library and embedded Content` and switch from `Do not Embed` to `Embed and sign` the app works.

<img width="572" alt="image" src="https://github.com/apache/cordova/assets/126869614/87bc0ed2-ccfe-4bcc-90c7-6d28bf732f99">

So, my question is how can I automate this step during my build process so i don't have to manually do it every time

**Comment by breautek** [comment, marked as answer]
I think you want to use the [framework](https://cordova.apache.org/docs/en/12.x/plugin_ref/spec.html#framework) tag instead.

I have an example usage which does exactly what you want in an older plugin that I used to maintain: https://github.com/totalpaveinc/cordova-plugin-libsqlite/blob/main/plugin.xml#L22

It does require [improvements](https://github.com/apache/cordova-ios/pull/1266) added in cordova-ios 7. If I recall correctly, older versions it was impossible to embed AND link at the same time, because linking was always the inverse of embedding. The PR merged for 7.0.0 release effectively adds the `link` directive allowing you to both embed and link. Therefore for this to work properly you'll also need to be using `cordova-ios@^7`.

**Reply by majeed-platogo** [reply]
This was the solution, Thank you so much
